packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -125,167 +91,96 @@ contract TaikoL2 is EssentialContract, TaikoL2Signer, ICrossChainSync {
-
-        // Check EIP-1559 basefee
+        // Verify the base fee per gas is correct
         uint256 basefee;
-        EIP1559Config memory config = getEIP1559Config();
-        if (config.gasIssuedPerSecond != 0) {
-            (basefee, gasExcess) = _calcBasefee({
-                config: config,
-                timeSinceParent: block.timestamp - parentTimestamp,
-                parentGasUsed: parentGasUsed
-            });
-        }
-
-        // On L2, basefee is not burnt, but sent to a treasury instead.
-        // The circuits will need to verify the basefee recipient is the
-        // designated address.
-        if (block.basefee != basefee) {
+        (basefee, gasExcess) = _calc1559BaseFee(config, l1Height, parentGasUsed);
+        if (!skipFeeCheck() && block.basefee != basefee) {
             revert L2_BASEFEE_MISMATCH();
         }
 
-        parentTimestamp = uint64(block.timestamp);
-
-        // We emit this event so circuits can grab its data to verify block
-        // variables.
-        // If plonk lookup table already has all these data, we can still use
-        // this event for debugging purpose.
-        emit Anchored({
-            number: uint64(block.number),
-            basefee: uint64(basefee),
-            gaslimit: uint32(block.gaslimit),
-            timestamp: uint64(block.timestamp),
-            parentHash: parentHash,
-            prevrandao: block.prevrandao,
-            coinbase: block.coinbase,
-            chainid: uint64(block.chainid)
-        });
-    }
-
-    /// @notice Updates EIP-1559 configurations.
-    /// @param _param1559 EIP-1559 parameters to set up the gas pricing model.
-    function updateEIP1559Config(EIP1559Params calldata _param1559)
-        public
-        onlyOwner
-    {
-        if (_param1559.gasIssuedPerSecond == 0) {
-            delete eip1559Config;
-            delete gasExcess;
-        } else {
-            if (
-                _param1559.basefee == 0 || _param1559.gasExcessMax == 0
-                    || _param1559.gasTarget == 0 || _param1559.ratio2x1x == 0
-            ) revert L2_INVALID_1559_PARAMS();
-
-            (uint128 xscale, uint128 yscale) = Lib1559Math.calculateScales({
-                xExcessMax: _param1559.gasExcessMax,
-                price: _param1559.basefee,
-                target: _param1559.gasTarget,
-                ratio2x1x: _param1559.ratio2x1x
-            });
-
-            if (xscale == 0 || xscale >= type(uint64).max || yscale == 0) {
-                revert L2_INVALID_1559_PARAMS();
-            }
-            eip1559Config.yscale = yscale;
-            eip1559Config.xscale = uint64(xscale);
-            eip1559Config.gasIssuedPerSecond = _param1559.gasIssuedPerSecond;
-
-            gasExcess = _param1559.gasExcessMax / 2;
+        // Store the L1's signal root as a signal to the local signal service to
+        // allow for multi-hop bridging.
+        if (l1SignalRoot != 0) {
```

Comment:
Does this mean that we allow L1 signal service undeployed by default? (If so, should we also add L1 signal service address to public input?)

Reply:
See https://github.com/taikoxyz/taiko-mono/pull/15035